confmap/expand.go

```diff
@@ -121,16 +128,27 @@ func (mr *Resolver) findAndExpandURI(ctx context.Context, input string) (any, bo
 	if err != nil {
 		return input, false, err
 	}
-	repl, err := toString(expanded)
+
+	var repl string
+	if internal.StrictlyTypedInputGate.IsEnabled() {
+		repl, err = expanded.AsString()
+	} else {
+		repl, err = toString(expanded)
+	}
 	if err != nil {
 		return input, false, fmt.Errorf("expanding %v: %w", uri, err)
 	}
 	return strings.ReplaceAll(input, uri, repl), true, err
 }
 
 // toString attempts to convert input to a string.
-func toString(input any) (string, error) {
+func toString(ret *Retrieved) (string, error) {
 	// This list must be kept in sync with checkRawConfType.
+	input, err := ret.AsRaw()
+	if err != nil {
+		return "", err
```

Comment:
Is this covered by e2e tests as well?

Reply:
There is no way to hit this path today https://github.com/open-telemetry/opentelemetry-collector/blob/48af1b8ea63a3393e468368beed47ce3668e5c73/confmap/provider.go#L152